src/codeql.ts

```diff
@@ -256,40 +399,21 @@ export function getSarifOutputType(
 // Generates sarif from the given bqrs file, if query kind supports it
 async function generateSarif(
   codeql: string,
-  bqrs: string,
   nwo: string,
-  queryMetadata: QueryMetadata,
-  sarifOutputType: SarifOutputType,
   databaseName: string,
-  sourceLocationPrefix: string,
+  queryPackName: string,
   databaseSHA?: string,
 ): Promise<Sarif> {
-  const {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- we are explicitly excluding this since it's calculated separately
-    kind,
-    id: queryId,
-    ...passthroughQueryMetadata
-  } = queryMetadata;
-
   const sarifFile = path.join("results", "results.sarif");
   await exec(codeql, [
```

Comment:
Random thought: how does this CodeQL CLI operation know to only generate SARIF for the correct queries? Does it automatically exclude non-problem queries? What happens if we call this when there aren't any problem queries in the pack?

My thought is, do we need to do the metadata and compatible-query-kinds parsing ourselves? Could we call this command unconditionally and use the output from the this command to determine if there were any problem queries or not?

Reply:
That's a good point, I'll check. I had assumed it would fail but perhaps not.

Reply:
It might fail, but if it fails saying "no suitable queries found" then that could still be acceptable. 🤷🏼

Reply:
It looks like a SARIF files is created, but with 0 results (https://github.com/dsp-testing/qc-controller/actions/runs/7475835885/job/20344865094#step:10:1447). I don't think that's what we want though.

Reply:
Created internal issue to cover this.